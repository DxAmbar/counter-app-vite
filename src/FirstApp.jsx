import PropTypes from 'prop-types';

export const App = ({ title, subTitle, name }) => {
    return (
    <>
    <h1 data-testid="test-title">{ title }</h1>
    <p>{ subTitle }</p>
    <p>{ name }</p>
    </>
    )
}

App.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

App.defaultProps = {
    name: "Daniela Baeza",
   // title: "No hay título",
    subTitle: "No hay subtítulo",

}