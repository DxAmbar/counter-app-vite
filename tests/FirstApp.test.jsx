import { render } from "@testing-library/react";
import { App } from "../src/FirstApp";


describe('Pruebas en FirstApp', () => { 

    test('Debe hacer match con el snapshot', () => {

        const title = 'Hola pirinola';
        const { container } = render( <App title={ title }/> );

        expect(container).toMatchSnapshot();

    });

    
    test('Debe mostrar el tÃ­tulo en un h1', () => { 

        const title = 'Hola pirinola';
        const { container, getByText } = render( <App title={ title }/> );

        expect (getByText( title )).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title );

    });


});