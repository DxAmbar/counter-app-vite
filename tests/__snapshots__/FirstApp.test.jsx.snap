// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola pirinola
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Daniela Baeza
  </p>
</div>
`;
